src/Microsoft.Azure.WebJobs.Extensions.Kafka/Listeners/KafkaTopicScaler.cs

```diff
@@ -0,0 +1,247 @@
+            long highestPartitionLag = 0L;
+            long totalLag = 0L;
+            foreach (var topicPartition in allPartitions)
+            {
+                // This call goes to the server always which probably yields the most accurate results. It blocks.
+                // Alternatively we could use consumer.GetWatermarkOffsets() that returns cached values, without blocking.
+                var watermark = consumer.QueryWatermarkOffsets(topicPartition, operationTimeout);
+
+                var commited = ownedCommittedOffset.FirstOrDefault(x => x.Partition == topicPartition.Partition);
+                if (commited != null)
+                {
+                    long diff;
+                    if (commited.Offset == Offset.Unset)
+                    {
+                        diff = watermark.High.Value;
+                    }
+                    else
+                    {
+                        diff = watermark.High.Value - commited.Offset.Value;
+                    }
+
+                    totalLag += diff;
+
+                    if (diff > highestPartitionLag)
+                    {
+                        highestPartitionLag = diff;
+                        partitionWithHighestLag = topicPartition.Partition;
+                    }
+                }
+            }
+
+            if (partitionWithHighestLag != Partition.Any)
+            {
+                logger.LogInformation("Total lag in '{topic}' is {totalLag}, highest partition lag found in {partition} with value of {offsetDifference}", this.topicName, totalLag, partitionWithHighestLag.Value, highestPartitionLag);
+            }
+
+            return Task.FromResult(new KafkaTriggerMetrics(totalLag, allPartitions.Count));
+        }
+
+        public ScaleStatus GetScaleStatus(ScaleStatusContext context)
+        {
+            return GetScaleStatusCore(context.WorkerCount, context.Metrics?.OfType<KafkaTriggerMetrics>().ToArray());
+        }
+
+        public ScaleStatus GetScaleStatus(ScaleStatusContext<KafkaTriggerMetrics> context)
+        {
+            return GetScaleStatusCore(context.WorkerCount, context.Metrics?.ToArray());
+        }
+
+        private ScaleStatus GetScaleStatusCore(int workerCount, KafkaTriggerMetrics[] metrics)
+        {
+            var status = new ScaleStatus
+            {
+                Vote = ScaleVote.None,
+            };
+
+            const int NumberOfSamplesToConsider = 5;
+
+            // At least 5 samples are required to make a scale decision for the rest of the checks.
+            if (metrics == null || metrics.Length < NumberOfSamplesToConsider)
+            {
+                return status;
+            }
+
+            var lastMetrics = metrics.Last();
+            long totalLag = lastMetrics.TotalLag;
+            long partitionCount = lastMetrics.PartitionCount;
+            long lagThreshold = 1000L;
+
+            //This case shouldn't ever happens as we do not scale out in case workerCount >= partitionCount
```

Comment:
It can happen - this trigger is only one scale vote for the app. There might also be an http trigger for example causing scale out past this partition count. Recommend just leaving a comment here [as in EventHub0(https://github.com/Azure/azure-functions-eventhubs-extension/blob/daab3ae8a8e08dd971b8f9b19d52f09ef44af081/src/Microsoft.Azure.WebJobs.Extensions.EventHubs/Listeners/EventHubsScaleMonitor.cs#L303)